adam-core/src/main/scala/org/bdgenomics/adam/rdd/ShuffleRegionJoin.scala

```diff
@@ -263,6 +263,30 @@ private[rdd] case class ManualRegionPartitioner(partitions: Int) extends Partiti
   }
 }
 
+private class AppendableIterator[T] extends Iterator[T] {
+  var iterators: ListBuffer[Iterator[T]] = ListBuffer.empty
+
+  def append(iter: Iterator[T]) {
+    if (iter.hasNext) {
+      iterators += iter
```

Comment:
In general, Scala has a pretty strict "advisory" (not quite a contract?) on using [Iterators](http://www.scala-lang.org/api/2.10.3/index.html#scala.collection.Iterator):

> It is of particular importance to note that, unless stated otherwise, one should never use an iterator after calling a method on it.

This is a private class used only in this file, so I think it is OK to make the assumption that `iter` is not modified after being passed.